docs/decisions/0008-application_state_differenciation.rst

```diff
@@ -0,0 +1,176 @@
+      enterpriseFeatures: PropTypes.shape({
+        topDownAssignmentRealTimeLcm: PropTypes.bool,
+      }).isRequired,
+      enablePortalLearnerCreditManagementScreen: PropTypes.bool.isRequired,
+    };
+
+    export default connect(mapStateToProps)(TestComponent);
+
+When writing tests for a hook that leverages useQuery, certain boilerplate code is required. Below is an example of how writing tests with a useQuery directly in the component in the form of a hook,
+where the ``props`` come from the ``portalConfiguration``.
+
+Note: the ``queryClient`` is a test utility function that populates the ``QueryClientProvider`` client prop. See its implementation `here <https://github.com/openedx/frontend-app-admin-portal/blob/c67c5e4d8a0328fe75cb9d46791a8b733fad8257/src/components/test/testUtils.jsx#L47-L56>`_
+
+.. code-block:: javascript
+
+    import { QueryClientProvider } from '@tanstack/react-query';
+    import { queryClient } from '../../test/testUtils';
+    import { Provider } from 'react-redux';
+    import configureMockStore from 'redux-mock-store';
+    import thunk from 'redux-thunk';
+    import {
+      screen,
+      render,
+    } from '@testing-library/react';
+    import '@testing-library/jest-dom/extend-expect';
+
+
+    // Configure the portal configuration used in the Provider
+    const mockStore = configureMockStore([thunk]);
+    const getMockStore = store => mockStore(store);
+    const initialStore = {
+      portalConfiguration: {
+        enterpriseId: 'test-id',
+        enterpriseFeatures: {
+          topDownAssignmentRealTimeLcm: true,
+        },
+        enablePortalLearnerCreditManagementScreen: true,
+      },
+    };
+
+    // Mock the hook that calls useQuery
+    jest.mock('../../EnterpriseSubsidiesContext/data/hooks', () => ({
+      ...jest.requireActual('../../EnterpriseSubsidiesContext/data/hooks'),
+      useEnterpriseBudgets: jest.fn().mockReturnValue({
+        data: {
+          budgets: [{
+            title: 'Test Budget',
+          },
+          ],
+        },
+      }),
+    }));
+
+    // Create your component wrapper
+    const TestComponentWrapper = () => (
+      <QueryClientProvider client={queryClient()}>
+        <Provider store={store}>
+            <TestComponent />
+        </Provider>
+      </QueryClientProvider>
+    );
+
+    describe('<TestComponent />', () => {
+      it('No budgets for your organization', () => {
+        // override the mocked value above to test an empty state for a useQuery response
+        useEnterpriseBudgets.mockReturnValueOnce(
+          { data: { budgets: [] } },
+        );
+        render(<TestComponentWrapper />);
+        expect(screen.getByText('No Budgets'));
+      });
+      it('budgets for your organization', () => {
+        render(<TestComponentWrapper />);
+        expect(screen.getByText('Test Budget'));
+      });
+    });
+
+
+
+
+
+Alternatives Considered
+***********************
+
+The alternative
```

Comment:
Might mention:
* disabling/modifying of React Query's refetch behavior (i.e., remove refetches altogether, extend `staleTime`).
* Attempting to migrate to context selectors (i.e., `use-context-selectors`) to mitigate re-rendering concerns.